commit 2a0a4d594b67c86565f1fe14783a70d62405ab0e
Author: Juergen Gross <jgross@suse.com>
Date:   Fri May 26 12:36:47 2017 +0100

    efi: Don't issue error message when booted under Xen
    
    commit 1ea34adb87c969b89dfd83f1905a79161e9ada26 upstream.
    
    When booted as Xen dom0 there won't be an EFI memmap allocated. Avoid
    issuing an error message in this case:
    
      [    0.144079] efi: Failed to allocate new EFI memmap
    
    Signed-off-by: Juergen Gross <jgross@suse.com>
    Signed-off-by: Matt Fleming <matt@codeblueprint.co.uk>
    Cc: Ard Biesheuvel <ard.biesheuvel@linaro.org>
    Cc: Linus Torvalds <torvalds@linux-foundation.org>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: Thomas Gleixner <tglx@linutronix.de>
    Cc: linux-efi@vger.kernel.org
    Link: http://lkml.kernel.org/r/20170526113652.21339-2-matt@codeblueprint.co.uk
    Signed-off-by: Ingo Molnar <mingo@kernel.org>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

diff --git a/arch/x86/platform/efi/quirks.c b/arch/x86/platform/efi/quirks.c
index cdfe8c6..393a0c0 100644
--- a/arch/x86/platform/efi/quirks.c
+++ b/arch/x86/platform/efi/quirks.c
@@ -358,6 +358,9 @@ void __init efi_free_boot_services(void)
 		free_bootmem_late(start, size);
 	}
 
+	if (!num_entries)
+		return;
+
 	new_size = efi.memmap.desc_size * num_entries;
 	new_phys = efi_memmap_alloc(num_entries);
 	if (!new_phys) {
